src/main/kotlin/uk/gov/justice/digital/hmpps/courthearingeventreceiver/service/MessageNotifier.kt

```diff
@@ -21,18 +22,32 @@ class MessageNotifier(
   @Value("\${aws.sns.topic_arn}")
   private val topicArn: String
 ) {
-  fun send(hearingEvent: HearingEvent) {
+  fun send(telemetryEventType: TelemetryEventType, hearingEvent: HearingEvent) {
 
-    val messageValue = MessageAttributeValue()
+    val messageTypeValue = MessageAttributeValue()
       .withDataType("String")
       .withStringValue(MESSAGE_TYPE)
 
+    val hearingEventTypeValue = MessageAttributeValue()
+      .withDataType("String")
+      .withStringValue(getHearingEventTypeValue(telemetryEventType))
+
     val publishRequest = PublishRequest(topicArn, objectMapper.writeValueAsString(hearingEvent))
-      .withMessageAttributes(mapOf("messageType" to messageValue))
+      .withMessageAttributes(mapOf("messageType" to messageTypeValue, "hearingEventType" to hearingEventTypeValue))
     val publishResult = amazonSNSClient.publish(publishRequest)
     log.info("Published message with message Id {}", publishResult.messageId)
   }
 
+  fun getHearingEventTypeValue(telemetryEventType: TelemetryEventType): String? {
```

Comment:
This feels like it ought to be the other way round, i.e. we take a HearingEventType and use that to work out the TelemetryEvent type 🤔 If we define a TelemetryEventType for each HearingType then we can get rid of the UNKNOWN type as well.

```
val type = HearingEventType.RESULTED
val telemetryType = type.getTelemetryEventType()
```

So in the controller it would look more like this:
```
    trackAndSendEvent(HearingEventType.RESULTED, hearingEvent)
...
private fun trackAndSendEvent(hearingEventType: HearingEventType, hearingEvent: HearingEvent) {
telemetryService.trackEvent(
      hearingEventType.getTelemetryEventType(),
      mapOf(
        "courtCode" to courtCode,
        "hearingId" to hearing.id
      )
```

What do you think?